chia/daemon/keychain_proxy.py

```diff
@@ -304,14 +304,14 @@ async def connect_to_keychain(
     ssl_context: Optional[ssl.SSLContext],
     log: logging.Logger,
     user: str = None,
-    testing: bool = False,
+    service: str = None,
 ) -> KeychainProxy:
     """
     Connect to the local daemon.
     """
 
     client = KeychainProxy(
-        uri=f"wss://{self_hostname}:{daemon_port}", ssl_context=ssl_context, log=log, user=user, testing=testing
+        uri=f"wss://{self_hostname}:{daemon_port}", ssl_context=ssl_context, log=log, user=user, service=service
```

Comment:
A substantial portion of the keyring/keychain refactoring work centered around having the daemon be the only entity that touches the keyring. That said, to prevent everything from breaking, the other processes still require access to the keys, but do so indirectly now (via KeychainProxy). The longer term goal is to separate out pubkeys and provide an interface by which the daemon can perform private key operations without exposing the key. So at some point we won't have private key material traveling over the wire.

At present, the websocket security is the same used for RPC between the daemon and other Chia processes.

Reply:
I see. it makes sense to have a service be responsible for the key. This authentication is client certificates + verify peers in SSL, right?

What makes me especially worry about this is that we use websockets, which was specifically designed to prevent web browsers to access arbitrary services, not meant for the web. And here we are developing such service and open it up to browsers.

Reply:
Yup, we're relying on the existing SSL setup to limit comms between chia processes.